### PR TITLE
Fix test for load-after-store in process-services

### PR DESCRIPTION
**Pull request by abstratt**

### CODE REVIEW

File: platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy [hunk 1]
```diff
@@ -35,11 +35,13 @@ class MemoryStatusUpdateIntegrationTest extends AbstractIntegrationSpec {
     }
 
     private static String waitForMemoryEventsTask() {
-        return '''
+        return groovyScript('''
             import java.util.concurrent.CountDownLatch
             import org.gradle.process.internal.health.memory.*
 
             task waitForMemoryEvents {
+                def projectDir = project.layout.projectDirectory
```
Comment (abstratt) [inline platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy:43]: These two lines were the only actually required

**Review by 6hundreds — APPROVED** [review]

File: platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy [hunk 2]
```diff
@@ -35,11 +35,13 @@ class MemoryStatusUpdateIntegrationTest extends AbstractIntegrationSpec {
...
+                def logger = logger
```
Comment (mlopatkin) [inline platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy:44]: :thought_balloon: I wonder why the `logger` reference is problematic, though. It should be resolved to `Task.getLogger` inside `doLast` clause, but, apparently, it only does so at `logger.warn "Waiting for memory status events..."` and not inside the callbacks, which have `BrokenScript` as `this` and `owner`. I don't fully understand the way Groovy resolves things at this point :(

**Review by mlopatkin — APPROVED** [review]

File: platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy (hunk 2 above)
Comment (abstratt) [reply platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy:44]: I had similar questions. No clue.

**Comment by abstratt** [comment]
@bot-gradle test and merge

File: platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy (hunk 2 above)
Comment (bamboo) [reply platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy:44]: Sounds like something we should fix as part of the adoption track: a `logger` reference within a `Closure` given as a `doLast` argument should resolve to `task.logger` without errors.

**Pull request merged**